src/main.rs

```diff
@@ -30,8 +39,44 @@ fn get_entry_field() -> gtk::Entry {
 
 #[allow(dead_code)]
 fn main() {
+    // Create a path to the desired file
+    let config_directory = env::home_dir().unwrap().join(Path::new(".config/rrun"));
+    fs::create_dir_all(&config_directory);
+    let config_path = config_directory.join(Path::new("config.toml"));
+    let config_path_display = config_path.display();
+
+    let mut file = match File::open(&config_path) {
+        Err(why) => {
+            println!("couldn't open {}: {}", config_path_display, Error::description(&why));
+            println!("Creating initial config file");
+            let mut f = File::create(&config_path).unwrap();
+            f.write_all(include_str!("config.toml").as_bytes()).unwrap();
+            panic!("Check the generated file at {} and try running again", config_path.display());
+        },
+        Ok(file) => file,
+    };
+
+    // Read the file contents into a string, returns `io::Result<usize>`
+    let mut toml = String::new();
+    match file.read_to_string(&mut toml) {
+        Err(why) => panic!("couldn't read {}: {}", config_path_display,
+                                                   Error::description(&why)),
+        Ok(_) => (),
+    }
+
+    let value = toml::Parser::new(&toml).parse().unwrap();
+    println!("TOML is {:?}", value);
+    let maybe_completions = value.get("completion").into_iter();
+    let completions = maybe_completions.flat_map(|cs| cs.as_slice().unwrap().into_iter());
     gtk::init().unwrap_or_else(|_| panic!("Failed to initialize GTK."));
     println!("Major: {}, Minor: {}", gtk::get_major_version(), gtk::get_minor_version());
+    let autocompleter_configs = completions.flat_map(|cs| cs.as_table());
```

Comment:
I think the configuration handling should probably bein a seperate function as to not increase the size of main() too much...